src/ui/viewmodels/MemoryViewerViewModel.cpp

```diff
@@ -1314,6 +1317,33 @@ void MemoryViewerViewModel::RenderHeader()
     }
 }
 
+void MemoryViewerViewModel::SaveToMemViewHistory()
+{
+    if (*s_iterMemViewHistoryIndex != GetAddress())
+    {
+        s_listMemViewHistory.erase(s_listMemViewHistory.begin(), s_iterMemViewHistoryIndex);
+        if (s_listMemViewHistory.size() > MEMVIEW_HISTORY_MAX_SIZE)
+            s_listMemViewHistory.pop_back();
+        s_listMemViewHistory.push_front(GetAddress());
```

Comment:
Recommend pushing new items on the back to avoid reallocating data until the history is full.